### PR TITLE
[Enhancement] Add an option to control whether to show progress bar in BaseInference	

### PR DESCRIPTION
**Pull request by W-ZN**
## Motivation

Provide solutions for issue #1112  solving.

## Modification

Added a bool parameter 'show_progress' to the 'BaseInference' class to control whether to show the progress bar, default is 'True', show the progress bar.

### CODE REVIEW

File: mmengine/infer/infer.py [hunk 1]
```diff
@@ -139,7 +139,8 @@ def __init__(self,
                  model: Union[ModelType, str, None] = None,
                  weights: Optional[str] = None,
                  device: Optional[str] = None,
-                 scope: Optional[str] = None) -> None:
+                 scope: Optional[str] = None,
+                 show_progress: bool = True) -> None:
```
Comment (HAOCHENYE) [inline mmengine/infer/infer.py:143]: Please update the docstring

Comment (W-ZN) [reply mmengine/infer/infer.py:145]: > Please update the docstring

Sorry, I forgot to modify the relevant text. Now I added a description of the parameters in the file

File: mmengine/infer/infer.py [hunk 2]
```diff
@@ -123,6 +123,8 @@ class BaseInferencer(metaclass=InferencerMeta):
         device (str, optional): Device to run inference. If None, the available
             device will be automatically used. Defaults to None.
         scope (str, optional): The scope of the model. Defaults to None.
+        show_progress (bool, optional): Control whether to display the progress bar
+            during the inference process. Defaults to True.
```
Comment (zhouzaida) [inline mmengine/infer/infer.py:127]:
```suggestion
        show_progress (bool): Control whether to display the progress bar during
            the inference process. Defaults to True.
```

**Review by zhouzaida — APPROVED** [review]

**Review by zhouzaida — APPROVED** [review]

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @W-ZN,

We'd like to express our appreciation for your valuable contributions to the mmengine. Your efforts have significantly aided in enhancing the project's quality.
It is our pleasure to invite you to join our community thorugh Discord_Special Interest Group (SIG) channel. This is a great place to share your experiences, discuss ideas, and connect with other like-minded people. To become a part of the SIG channel, send a message to the moderator, OpenMMLab, briefly introduce yourself and mention your open-source contributions in the #introductions channel. Our team will gladly facilitate your entry. We eagerly await your presence. Please follow this link to join us: ​https://discord.gg/UjgXkPWNqA.

If you're on WeChat, we'd also love for you to join our community there. Just add our assistant using the WeChat ID: openmmlabwx. When sending the friend request, remember to include the remark "mmsig + Github ID".

Thanks again for your awesome contribution, and we're excited to have you as part of our community!

**Pull request merged**